web/modules/custom/weather_block/src/Plugin/Block/WeatherBlock.php

```diff
@@ -111,17 +130,10 @@ public function getCacheTags(): array {
    * {@inheritdoc}
    */
   public function blockForm($form, FormStateInterface $form_state) {
-    $cities = [
-      'Вінниця', 'Дніпро', 'Донецьк', 'Житомир', 'Запоріжжя', 'Івано-Франківськ',
-      'Київ', 'Кропивницький', 'Луганськ', 'Луцьк', 'Львів', 'Миколаїв', 'Одеса',
-      'Полтава', 'Рівне', 'Суми', 'Тернопіль', 'Ужгород', 'Харків', 'Херсон',
-      'Хмельницький', 'Черкаси', 'Чернівці', 'Чернігів', 'Сімферополь',
-    ];
+    $cities = $this->cityService->getCitiesArray();
 
-    $user = \Drupal::currentUser();
-    $user_id = $user->id();
-
-    $user_config = $this->configuration['users'][$user_id] ?? [];
+    $user_id = $this->accountProxy->id();
+    $savedCities = $this->cityService->getSavedCityForUser($user_id);
```

Comment:
Why user-level setting is in blockForm? User's should change their city from user city form